src/grid_surf.cpp

```diff
@@ -1976,20 +1990,30 @@ int Grid::point_outside_surfs_explicit(int icell, double *x)
         if (face == 6 and norm[2] > 0.0) continue;
       }
 
-      double onethird = 1.0/3.0;
-      x[0] = onethird*(cpath[0]+cpath[3]+cpath[6]) + displace*norm[0];
-      x[1] = onethird*(cpath[1]+cpath[4]+cpath[7]) + displace*norm[1];
-      x[2] = onethird*(cpath[2]+cpath[5]+cpath[8]) + displace*norm[2];
-      return 1;
+      // use the surface with the largest clipped surface area
+      // for surfaces with a tiny intersection, the point can get pushed
```

Comment:
for surfaces with a tiny intersection, the point to push off from can be very close to a shared edge
  with another triangle - if the angle between the two tris is less than 90 degrees, the pushed-off
  point can end up "inside" the surface instead of "outside"
using the triangle with the largest clipped area and pushing off its centroid makes this
  issue very unlikely to occur